core/src/main/java/io/grpc/ClientCallImpl.java

```diff
@@ -85,27 +108,21 @@
     ClientTransport get();
   }
 
+  private enum State {INIT, STARTED, HALF_CLOSED, CANCELED}
+
   ClientCallImpl<ReqT, RespT> setUserAgent(String userAgent) {
     this.userAgent = userAgent;
     return this;
   }
 
   @Override
-  public void start(Listener<RespT> observer, Metadata.Headers headers) {
-    Preconditions.checkState(stream == null, "Already started");
+  public void start(ClientCall.Listener<RespT> observer, Metadata.Headers headers) {
```

Comment:
It's so hard to tell which oneliners are a big deal and which ones aren't. :P 

Also, Done.